packages/xo-server/src/xapi/mixins/storage.mjs

```diff
@@ -74,17 +78,22 @@ export default {
     const vdis = this.getObject(sr).$VDIs
     const unhealthyVdis = { __proto__: null }
     const children = groupBy(vdis, 'sm_config.vhd-parent')
+    const vdisWithUnknownVhdParent = []
+
     const cache = { __proto__: null }
     forEach(vdis, vdi => {
       if (vdi.managed && !vdi.is_a_snapshot) {
         const { uuid } = vdi
-        const length = this._getUnhealthyVdiChainLength(uuid, children, cache)
+        const length = this._getUnhealthyVdiChainLength(uuid, children, cache, vdisWithUnknownVhdParent)
         if (length !== 0) {
           unhealthyVdis[uuid] = length
         }
       }
     })
-    return unhealthyVdis
+    return {
+      vdisWithUnknownVhdParent,
+      unhealthyVdis,
+    }
```

Comment:
This breaks the API, right? Maybe we should create a new one instead of trying to fit everything in this one.

Reply:
I think it's not a big deal because it really is an internal API, and it makes sense to have both these features grouped:

- the same code is used for detection of these
- it allows xo-web to have a single subscription for this